pkg/logs/pipeline/pipeline.go

```diff
@@ -17,36 +19,37 @@ import (
 type Pipeline struct {
 	InputChan chan *message.Message
 	processor *processor.Processor
-	sender    *sender.Sender
+	sender    sender.Sender
 }
 
 // NewPipeline returns a new Pipeline
-func NewPipeline(outputChan chan *message.Message, processingRules []*config.ProcessingRule, endpoints *client.Endpoints, destinationsContext *client.DestinationsContext) *Pipeline {
-	// initialize the main destination
-	main := client.NewDestination(endpoints.Main, destinationsContext)
-
-	// initialize the additional destinations
-	var additionals []*client.Destination
-	for _, endpoint := range endpoints.Additionals {
-		additionals = append(additionals, client.NewDestination(endpoint, destinationsContext))
-	}
-
+func NewPipeline(outputChan chan *message.Message, processingRules []*config.ProcessingRule, endpoints *config.Endpoints, destinationsContext *tcp.DestinationsContext) *Pipeline {
 	// initialize the sender
-	destinations := client.NewDestinations(main, additionals)
+	destinations := client.NewDestinations(endpoints, destinationsContext)
 	senderChan := make(chan *message.Message, config.ChanSize)
-	sender := sender.NewSender(senderChan, outputChan, destinations)
+	var newsender sender.Sender
+	if coreConfig.Datadog.GetBool("logs_config.use_http") {
```

Comment:
nits: I think we could merge sender and encoder constructions inside the same block